jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java

```diff
@@ -492,11 +506,16 @@ public void signalEvent(String type, Object event) {
 			        }
 				}
 				if (((org.jbpm.workflow.core.WorkflowProcess) getWorkflowProcess()).isDynamic()) {
-					for (Node node : getWorkflowProcess().getNodes()) {
+				    Node [] nodes = getWorkflowProcess().getNodes();
+				    // OCRAM: UGLY
+				    boolean stackless = isStackless();
```

Comment:
agreed :) could you please elaborate a bit more why this loop with limits is needed?